src/gsy_e/models/market/forward.py

```diff
@@ -129,44 +142,58 @@ def _get_end_time(current_time: DateTime) -> DateTime:
     def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
         return duration(minutes=15)
 
-    def _calculate_closing_time(self, delivery_time: DateTime) -> DateTime:
-        """Closing time of the intraday market is 15 mins before delivery."""
-        return delivery_time - duration(minutes=15)
+    @property
+    def _time_from_market_close_till_delivery(self) -> duration:
+        return duration(minutes=15)
```

Comment:
Since the matter was mentioned once more, let me retry the approach that we had before my PR, where rotation was controlled by the closing time of each market (therefore implicitly by the time_from_market_close_till_delivery). I think this will simplify the process

Reply:
@spyrostz I think this wasn't addressed in the recent commits, right? Do you plan on working on it or is the PR complete?

Reply:
@fievelk thanks for the reminder, I'll try to address it now so that it is not forgotten!